### ISSUE TITLE
ngx_pagespeed build locally but not on PPA

### ISSUE THREAD

**Issue by Napsty** [pagespeed/ngx_pagespeed#1301]
I created a nginx-pagespeed package for my PPA and I hit something strange.
While debuild works fine on my local machine, on the PPA server it always fails with:

	adding module in /«PKGBUILDDIR»/debian/modules/pagespeed
	mod_pagespeed_dir=/«PKGBUILDDIR»/debian/modules/pagespeed/psol/include
	build_from_source=false
	checking for psol ... not found
	./configure: error: module ngx_pagespeed requires the pagespeed optimization library.
	Look in obj/autoconf.err for more details.
	debian/rules:60: recipe for target 'override_dh_auto_configure' failed
	make[1]: *** [override_dh_auto_configure] Error 1
	make[1]: Leaving directory '/«PKGBUILDDIR»'
	debian/rules:57: recipe for target 'build-arch' failed
	make: *** [build-arch] Error 2
	dpkg-buildpackage: error: debian/rules build-arch gave error exit status 2

but psol exists, otherwise local build wouldn't work (here output from local binary package build):

	adding module in /home/claudio/nginx-1.10.1/debian/modules/pagespeed
	mod_pagespeed_dir=/home/claudio/nginx-1.10.1/debian/modules/pagespeed/psol/include
	build_from_source=false
	checking for psol ... found
	List of modules (in reverse order of applicability): ngx_http_write_filter_module ngx_http_header_filter_module ngx_http_chunked_filter_module ngx_http_v2_filter_module ngx_http_range_header_filter_module ngx_pagespeed_etag_filter ngx_http_gzip_filter_module ngx_pagespeed ngx_http_postpone_filter_module ngx_http_ssi_filter_module ngx_http_charset_filter_module ngx_http_xslt_filter_module ngx_http_image_filter_module ngx_http_sub_filter_module ngx_http_addition_filter_module ngx_http_gunzip_filter_module ngx_http_userid_filter_module ngx_http_headers_filter_module ngx_http_subs_filter_module
	checking for psol-compiler-compat ... found
	 + ngx_pagespeed was configured

I also verified that psol is included in the package sent to the PPA server with dput:

	tar tf nginx-pagespeed_1.10.1-20161021.tar.xz
	[...]
	nginx-1.10.1/debian/modules/pagespeed/
	nginx-1.10.1/debian/modules/pagespeed/test/
	nginx-1.10.1/debian/modules/pagespeed/test/nginx_system_test.sh
	nginx-1.10.1/debian/modules/pagespeed/test/run_tests.sh
	nginx-1.10.1/debian/modules/pagespeed/test/valgrind.sup
	nginx-1.10.1/debian/modules/pagespeed/test/pagespeed_test.conf.template
	nginx-1.10.1/debian/modules/pagespeed/LICENSE
	nginx-1.10.1/debian/modules/pagespeed/config
	nginx-1.10.1/debian/modules/pagespeed/README.md
	nginx-1.10.1/debian/modules/pagespeed/cpp_feature
	nginx-1.10.1/debian/modules/pagespeed/config.make
	nginx-1.10.1/debian/modules/pagespeed/psol/
	nginx-1.10.1/debian/modules/pagespeed/psol/lib/
	nginx-1.10.1/debian/modules/pagespeed/psol/lib/Debug/
	nginx-1.10.1/debian/modules/pagespeed/psol/lib/Debug/linux/
	nginx-1.10.1/debian/modules/pagespeed/psol/lib/Debug/linux/x64/
	nginx-1.10.1/debian/modules/pagespeed/psol/lib/Debug/linux/ia32/
	nginx-1.10.1/debian/modules/pagespeed/psol/lib/Release/
	nginx-1.10.1/debian/modules/pagespeed/psol/lib/Release/linux/
	nginx-1.10.1/debian/modules/pagespeed/psol/lib/Release/linux/x64/
	nginx-1.10.1/debian/modules/pagespeed/psol/lib/Release/linux/x64/pagespeed_js_minify
	nginx-1.10.1/debian/modules/pagespeed/psol/lib/Release/linux/ia32/
	nginx-1.10.1/debian/modules/pagespeed/psol/lib/Release/linux/ia32/pagespeed_js_minify
	nginx-1.10.1/debian/modules/pagespeed/psol/include/
	nginx-1.10.1/debian/modules/pagespeed/psol/include/net/
	[...]

Any ideas why this fails on PPA?

**Comment by morlovich** [comment]
Any chances you could get your hands on that obj/autoconf.err file? 
Basically it tried to link a test program against psol.a (actually can't see if you have that file or not due to the [...] --- do you?), and the obj/autoconf.err contains the error messages.

**Comment by Napsty** [comment]
No, unfortunately not. I'd love to ;) However I think this is just a build-container starting up to build to package on the PPA server and then distroys itself. The only thing I see is the build log. 
One way would be to make a "cat obj/autoconf.err" somewhere, maybe in the debian/rules file? But I'm not sure if that will work. Gotta try.

You can find the build log here, as long as I keep the PPA (it's just a test anyway): https://launchpadlibrarian.net/290276460/buildlog_ubuntu-xenial-amd64.nginx-pagespeed_1.10.1-20161021_BUILDING.txt.gz 

Nope, "psol.a" does not exist. Just checked. Doesn't this come with the psol tar package? Do I have to download additional psol libraries/files?

**Comment by Napsty** [comment]
I was now able to get my hands on "obj/autoconf.err" with the following trick in the pagespeed config:

	if [ $ngx_found = no ]; then
	  cat << END
	$0: error: module ngx_pagespeed requires the pagespeed optimization library.
	Look in obj/autoconf.err for more details.
	$(pwd)
	$(find . -name 'autoconf.err' -exec cat {} +)
	END

Here's the relevant part of autoconf.err:

	----------------------------------------
	checking for psol

	cc: error: /«PKGBUILDDIR»/debian/modules/pagespeed/psol/lib/Debug/linux/x64/pagespeed_automatic.a: No such file or directory

That file exists in my local source folder, but it seems it was not packed in the tar.xz file! No wonder it doesn't turn up on the PPA server. I know now this is not a pagespeed fault, but any idea why this file was not packaged by debuild?

**Comment by morlovich** [comment]
Sorry, correction: pagespeed_automatic.a
(something like: psol/lib/Release/linux/x64/pagespeed_automatic.a)

I don't know which files you can easily edit based on how it's setup. Our
"config" file would be a potential spot, since it's what produces that
error message at this fragment:
if [ $ngx_found = no ]; then
  cat << END
$0: error: module ngx_pagespeed requires the pagespeed optimization library.
Look in $PWD/$NGX_AUTOCONF_ERR for more details.
END
  exit 1
fi

**Comment by morlovich** [comment]
I think that's talking about the final package... or I guess they expect
the inputs to be either tarballs or other debian packages?

You could just build the entire thing from source, although that's a bit of
a waste of a CPU (and a lot more steps), but it is kinda nice to use the
system compiler for everything.
